### PR TITLE
Allow configuring max connections through config

### PR DESCRIPTION
**Pull request by carlhoerberg**
## Proposed Changes

Allow configuring max connections through `rabbitmq.config`, using Ranch's max_connections setting:
https://ninenines.eu/docs/en/ranch/1.4/guide/listeners/#_limiting_the_number_of_concurrent_connections

Today the only other way (without a LB) is to limit number of file descriptors, but we often want a higher FD limit than connection limit as queues consumes many FDs. We want to limit connections as misbehaving client might open thousands of connections and consume significant amount of RAM and even OOM the server. 

The change is non-breaking and defaults to `infinity` as today. 

Documentation, tests etc are not added. Wanted to  open up the discussion first. 

## Types of Changes

- [ ] Bugfix (non-breaking change which fixes issue #NNNN)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance)

## Checklist

- [ ] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [ ] All tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in related repositories

## Further Comments

If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@carlhoerberg Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/rabbitmq?repositoryId=rabbitmq/rabbitmq-server&pullRequestId=1400)!

[Click here](https://cla.pivotal.io/sync/rabbitmq?repositoryId=rabbitmq/rabbitmq-server&pullRequestId=1400) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by michaelklishin** [comment]
Seems reasonable and straightforward to add.

I think this can go into `stable`.

**Comment by pivotal-issuemaster** [comment]
@carlhoerberg Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/rabbitmq?repositoryId=rabbitmq/rabbitmq-server&pullRequestId=1400)!

File: src/tcp_listener_sup.erl [hunk 1]
```diff
@@ -49,10 +49,11 @@ start_link(IPAddress, Port, Transport, SocketOpts, ProtoSup, ProtoOpts, OnStartu
 init({IPAddress, Port, Transport, SocketOpts, ProtoSup, ProtoOpts, OnStartup, OnShutdown,
       ConcurrentAcceptorCount, Label}) ->
     {ok, AckTimeout} = application:get_env(rabbit, ssl_handshake_timeout),
+    MaxConnections = application:get_env(rabbit, max_connections, infinity),
```
Comment (michaelklishin) [inline src/tcp_listener_sup.erl]: `application:get_env/3` isn't available in Erlang/OTP R16B03. Please use `rabbit_misc:get_env/3`.

**Comment by michaelklishin** [comment]
@carlhoerberg I will rename the key to `rabbit.connection_max` (we already have `channel_max` and `frame_max`) and update the docs. Thank you.

**Comment by carlhoerberg** [comment]
Ok, thank you ! :)

**Pull request merged**